### PR TITLE
[NOD-1201] Panic if callbacks are not set

### PR DESCRIPTION
**Pull request by someone235**

### CODE REVIEW

File: netadapter/server/grpcserver/grpc_server.go [hunk 1]
```diff
@@ -37,6 +37,10 @@ func NewGRPCServer(listeningAddrs []string) (server.Server, error) {
 }
 
 func (s *gRPCServer) Start() error {
+	if s.onConnectedHandler == nil {
```
Comment (svarogg) [inline netadapter/server/grpcserver/grpc_server.go:40]: We agreed that only grpcConnection is allowed to panic immidiately, grpcServer can return error up to app.Start, which will panic for it.

File: netadapter/netconnection.go [hunk 2]
```diff
@@ -15,23 +15,28 @@ type NetConnection struct {
...
-		connection: connection,
-		router:     router,
+		connection:         connection,
+		router:             router,
+		invalidMessageChan: make(chan error),
 	}
 
 	netConnection.connection.SetOnDisconnectedHandler(func() {
 		router.Close()
+		close(netConnection.invalidMessageChan)
```
Comment (stasatdaglabs) [inline netadapter/netconnection.go:34]: Please create a Close function in netConnection and move this there instead.

File: protocol/protocolerrors/protocolerrors.go [hunk 3]
```diff
@@ -36,17 +35,15 @@ func New(shouldBan bool, message string) error {
 	}
 }
 
-// Wrap returns an error annotating err with a stack trace
-// at the point Wrap is called, and the supplied message.
+// Wrap wraps the given error and returns it as a ProtoclError.
```
Comment (stasatdaglabs) [inline protocol/protocolerrors/protocolerrors.go:38]: Typo: ProtoclError -> ProtocolError

File: protocol/protocolerrors/protocolerrors.go [hunk 4]
```diff
@@ -36,17 +35,15 @@ func New(shouldBan bool, message string) error {
...
 func Wrap(shouldBan bool, err error, message string) error {
 	return &ProtocolError{
 		ShouldBan: shouldBan,
 		Cause:     errors.Wrap(err, message),
 	}
 }
 
-// Wrapf returns an error annotating err with a stack trace
-// at the point Wrapf is called, and the format specifier.
+// Wrapf wraps the given error with the given format and returns it as a ProtoclError.
```
Comment (stasatdaglabs) [inline protocol/protocolerrors/protocolerrors.go:46]: Typo: ProtoclError -> ProtocolError

**Pull request merged**